hq_superset/migrations/env.py

```diff
@@ -0,0 +1,77 @@
+import urllib.parse
+from logging.config import fileConfig
+
+from alembic import context
+from flask import current_app
+from sqlalchemy import engine_from_config, pool
+
+from hq_superset.const import OAUTH2_DATABASE_NAME
+from hq_superset.models import HQClient
+
+config = context.config
+if config.config_file_name is not None:
+    fileConfig(config.config_file_name)
+
+db_uri = current_app.config['SQLALCHEMY_BINDS'][OAUTH2_DATABASE_NAME]
+decoded_uri = urllib.parse.unquote(db_uri)
+config.set_main_option('sqlalchemy.url', decoded_uri)
+
+# add your model's MetaData object here for 'autogenerate' support
+target_metadata = HQClient.metadata
+
+# other values from the config, defined by the needs of env.py,
+# can be acquired:
+# my_important_option = config.get_main_option("my_important_option")
+# ... etc.
+
+
+def run_migrations_offline() -> None:
+    """Run migrations in 'offline' mode.
+
+    This configures the context with just a URL
+    and not an Engine, though an Engine is acceptable
+    here as well.  By skipping the Engine creation
+    we don't even need a DBAPI to be available.
+
+    Calls to context.execute() here emit the given string to the
+    script output.
+
+    """
+    url = config.get_main_option("sqlalchemy.url")
+    context.configure(
+        url=url,
+        target_metadata=target_metadata,
+        literal_binds=True,
+        dialect_opts={"paramstyle": "named"},
+    )
+
+    with context.begin_transaction():
+        context.run_migrations()
+
+
+def run_migrations_online() -> None:
+    """Run migrations in 'online' mode.
+
+    In this scenario we need to create an Engine
+    and associate a connection with the context.
+
+    """
+    connectable = engine_from_config(
+        config.get_section(config.config_ini_section, {}),
+        prefix="sqlalchemy.",
+        poolclass=pool.NullPool,
+    )
+
+    with connectable.connect() as connection:
+        context.configure(
+            connection=connection, target_metadata=target_metadata
+        )
+
+        with context.begin_transaction():
+            context.run_migrations()
+
+
+if context.is_offline_mode():
+    run_migrations_offline()
+else:
+    run_migrations_online()
```

Comment:
Did we take this process from somewhere?
Just thinking why we would need offline/online processes.